common/src/main/java/tachyon/Version.java

```diff
@@ -25,7 +25,7 @@
 
   static {
     TachyonConf tachyonConf = new TachyonConf();
-    VERSION = tachyonConf.get(Constants.TACHYON_VERSION, "UNDEFINED");
+    VERSION = tachyonConf.get(Constants.TACHYON_VERSION);
```

Comment:
Precondition is probably not necessary, because most keys are defined in `tachyon-default.properties` so they should exist (or an exception is thrown). There are a few (<10) properties that are not defined in `tachyon-default.properties` because they can only be set on runtime. I checked the code base so that we always check existence before query the value. I am still thinking a better way to do this more systematically, but maybe out of the scope of this PR.